### PR TITLE
Add inchain applications configurations

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

Add applications that be triggered as in-chain application for the deployment-chain example instead of reuse the existed ones

```console
$ tree examples/deployment-chain       
examples/deployment-chain
├── chain-simple-cloudrun
│   ├── app.pipecd.yaml
│   └── service.yaml
├── chain-simple-lambda
│   ├── app.pipecd.yaml
│   └── function.yaml
└── simple
    ├── app.pipecd.yaml
    ├── deployment.yaml
    └── service.yaml

3 directories, 7 files
```

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: examples/deployment-chain/simple/simple-cloudrun/app.pipecd.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Quick sync by rolling out the new version and switching all traffic to it.
+apiVersion: pipecd.dev/v1beta1
+kind: CloudRunApp
+spec:
+  name: inchain-simple-cloudrun
```
Comment (nghialv) [inline examples/deployment-chain/simple/simple-cloudrun/app.pipecd.yaml:5]: To be unified with other examples, how about using the same name for directory and application name?

e.g. `chain-simple-cloudrun`

Comment (khanhtc1202) [reply examples/deployment-chain/simple/simple-cloudrun/app.pipecd.yaml]: Nice catch, addressed by https://github.com/pipe-cd/pipe/pull/2993/commits/11735b481165704333086f9dacc97912a6e99400 🙏

File: examples/deployment-chain/simple/chain-simple-lambda/function.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apiVersion: pipecd.dev/v1beta1
+kind: LambdaFunction
+spec:
+  name: InChainSimpleFunction
```
Comment (nghialv) [inline examples/deployment-chain/simple/chain-simple-lambda/function.yaml:4]:
```suggestion
  name: ChainSimpleLambda
```

File: examples/deployment-chain/simple/chain-simple-cloudrun/service.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: inchain-simple-cloudrun
```
Comment (nghialv) [inline examples/deployment-chain/simple/chain-simple-cloudrun/service.yaml:4]:
```suggestion
  name: chain-simple-cloudrun
```

File: examples/deployment-chain/simple/app.pipecd.yaml [hunk 4]
```diff
@@ -10,13 +10,13 @@ spec:
...
-        - name: simple
+        # https://github.com/pipe-cd/examples/blob/master/deployment-chain/simple/chain-simple-cloudrun
+        - name: chain-simple-cloudrun
           kind: CLOUDRUN
-        # The below will match application named `simple` of kind `LAMBDA`.
+        # The below will match application named `chain-simple-lambda` of kind `LAMBDA`.
         # which means the app that it's configuration is at
-        # https://github.com/pipe-cd/examples/tree/master/lambda/simple
-        - name: simple
+        # https://github.com/pipe-cd/examples/tree/master/deployment-chain/simple/chain-simple-lambda
```
Comment (nghialv) [inline examples/deployment-chain/simple/app.pipecd.yaml:20]: Putting all in-chain applications inside the application directory of the first one can lead to misunderstandings by our users.
They might think that only ones in that directory could be used in the chain.
So How about putting them at the same directory level?

Comment (khanhtc1202) [reply examples/deployment-chain/simple/app.pipecd.yaml]: Sure, I don't have a strong opinion on this 👍

**Comment by nakabonne** [comment]
👍 
/approve

**Pull request merged**